compiler/testData/codegen/bytecodeText/sam/samWrapperSyntheticFlags.kt

```diff
@@ -0,0 +1,23 @@
+// FILE: J.java
+
+public class J {
+    public static void g(Runnable r) {
+        r.run();
+    }
+}
+
+// FILE: test.kt
```

Comment:
Maybe put these two functions to separate `FILE`s, so that the file facade name in the pattern below (`TestKt` now) would clearly indicate which one is which?

Reply:
On second thought, maybe this test would even work better as a `bytecodeListing` test, instead of bytecode text?

Reply:
There is a bytecode listing test in `bytecodeListing/samAdapterAndInlinedOne.kt`, but it would need to have a separate expectation file for the IR backend, or alternatively fixes in the JVM backend. The JVM backend doesn't generate the correct inner class attributes and omits the nullability annotations on non-synthetic SAM constructors.

Actually I've added the expectation file for the IR backend now, since these are the only remaining differences between the backends and the IR backend is arguably more correct here.

Reply:
Thanks! Yes, the expectation file for JVM IR looks good. So now that we have it, can we add `// WITH_SIGNATURES
` to `bytecodeListing/samAdapterAndInlinedOne.kt` to also test generic signatures, and remove the new bytecode text tests?

Reply:
Sure! Done.